src/main/java/org/openstreetmap/atlas/geography/atlas/builder/proto/ProtoAtlasBuilder.java

```diff
@@ -0,0 +1,237 @@
+        try
+        {
+            protoAtlasContainer = ProtoAtlasContainer.parseFrom(resource.readBytesAndClose());
+        }
+        // TODO do something smarter here?
+        catch (final InvalidProtocolBufferException exception)
+        {
+            throw new CoreException("Error parsing the protobuf");
+        }
+
+        // initialize atlas metadata
+        final long numberOfPoints = protoAtlasContainer.getNumberOfPoints();
+        final long numberOfLines = protoAtlasContainer.getNumberOfLines();
+        final long numberOfAreas = 0;
+        final long numberOfNodes = 0;
+        final long numberOfEdges = 0;
+        final long numberOfRelations = 0;
+        final AtlasSize atlasSize = new AtlasSize(numberOfEdges, numberOfNodes, numberOfAreas,
+                numberOfLines, numberOfPoints, numberOfRelations);
+        final AtlasMetaData atlasMetaData = new AtlasMetaData(atlasSize, true, "unknown",
+                "ProtoAtlas", "unknown", "unknown", Maps.hashMap());
+        final PackedAtlasBuilder builder = new PackedAtlasBuilder().withSizeEstimates(atlasSize)
+                .withMetaData(atlasMetaData).withName(resource.getName());
+
+        // build the atlas features
+        parsePoints(builder, protoAtlasContainer.getPointsList());
+        parseLines(builder, protoAtlasContainer.getLinesList());
+
+        return (PackedAtlas) builder.get();
+    }
+
+    /**
+     * Write an Atlas to a resource in the naive ProtoAtlas format.
+     *
+     * @param atlas
+     *            the Atlas to be written
+     * @param resource
+     *            the resource to write into
+     */
+    public void write(final Atlas atlas, final WritableResource resource)
+    {
+        final ProtoAtlasContainer.Builder protoAtlasBuilder = ProtoAtlasContainer.newBuilder();
+
+        writePointsToBuilder(atlas, protoAtlasBuilder);
+        writeLinesToBuilder(atlas, protoAtlasBuilder);
+
+        final ProtoAtlasContainer protoAtlas = protoAtlasBuilder.build();
+        resource.writeAndClose(protoAtlas.toByteArray());
+    }
+
+    private List<ProtoTag> buildProtoTagListFromTagMap(final Map<String, String> tags)
+    {
+        final List<ProtoTag> protoTags = new ArrayList<>();
+        for (final Map.Entry<String, String> entry : tags.entrySet())
+        {
+            final ProtoTag.Builder tagBuilder = ProtoTag.newBuilder();
+
+            // TODO what happens if entry.getValue() returns null?
+            // this could happen if someone inserted a null entry at that key
+            // maybe use entry.getValue() != null ? entry.getValue() : ""
+            final String fullTag = entry.getKey() + TAG_DELIMITER + entry.getValue();
+            tagBuilder.setTagText(fullTag);
+            protoTags.add(tagBuilder.build());
+        }
+        return protoTags;
+    }
+
+    private ProtoLocation convertLocationToProtoLocation(final Location location)
```

Comment:
Check out the `Converter` and `TwoWayConverter` interfaces, it might be a good pattern to follow here as well. Create a proto conversion package that has all the converters (with convert and backwardConvert), then use them here in the builder.

Reply:
Sounds good! James actually mentioned the exact same thing.